### ISSUE TITLE
Add random spatial and intensity transforms

### ISSUE THREAD

**Issue by jordanpn** [eisen-ai/eisen-core#20]
It would be very helpful to have support for various spatial and intensity transforms, similar to what is already available in torchio transforms and MONAI. Supporting torchio in eisen-extras may be the most sensible way to go, imo.

The specific transforms I am interested in include:
- random affine (3D, 4D)
- random shift and crop (3D, 4D)
- random noise, ideally with specified distribution (3D, 4D)
- random flips (3D, 4D)

Note: 4D in my definition means supporting multichannel data (e.g. T1 and T2 channel) and applying the same 3D transform to each channel.

**Comment by fepegar** [comment]
Let me know if you need anything :)

**Comment by jordanpn** [comment]
Awesome, thanks!

**Comment by faustomilletari** [comment]
@fepegar would you be okay with having wrappers that facilitate use of TorchIo transforms into Eisen? I would ideally propose to place them in Eisen-extras and introduce TorchIo as a requirement for eisen-extras

**Comment by fepegar** [comment]
Yes, that sounds good. I'm also happy to adapt some aspects of the library if that would help integration.

**Comment by jordanpn** [comment]
Hi Fernando, just wanted to ask, did you make any progress on the wrapper implementation? I may take a stab at it if you haven't started yet.

**Comment by fepegar** [comment]
Hi @jordanpn, I've obviously misunderstood the plan here, I didn't know you were expecting me to work on this.

**Comment by faustomilletari** [comment]
Totally my fault here. I was supposed to get this done. I will try it tomorrow morning because I have been on vacation. Let’s get it done. I would love your help for review of the PR! Thanks for the patience

**Comment by jordanpn** [comment]
Thank you, I am happy to help with the PR review and testing!

**Comment by faustomilletari** [comment]
Thanks @jordanpn . @fepegar could we talk on slack or email? I might have a couple questions!

**Comment by faustomilletari** [comment]
This is the slack link https://join.slack.com/t/eisen-ai/shared_invite/enQtOTIwOTg0ODc2MDY3LWE5MGVlNTgxZTBmNjM5NGU2OWRhYzdiMzAzNzRmMGZlZWJlZmEzZDcyNWE4ZTQ5OGM4NDk5N2E5NTEyYzE3NGI

**Comment by faustomilletari** [comment]
Please refer to this PR https://github.com/eisen-ai/eisen-extras/pull/1

**Comment by RespectKnowledge** [comment]
Hi,
This is excellent framework. I have one question, How can I prepare json file for custom dataset to use your framework.
Thanks

**Comment by faustomilletari** [comment]
The typical form for a json dataset is a list of dictionaries. Each entry of the list contains a dictionary. In the dictionary you can store whatever you want. Typically we store paths to files. Eg. [{“image”: “path-to-image.jpg”, “label”: 7}, ...]
